tests/app/dao/test_fact_billing_dao.py

```diff
@@ -661,118 +649,124 @@ def test_fetch_sms_free_allowance_remainder_until_date_with_two_services(notify_
     assert service_2_result[0] == (service_2.id, 20, 22, 0)
 
 
-def test_fetch_usage_for_all_services_sms_for_first_quarter(notify_db_session):
-    # This test is useful because the inner query resultset is empty.
-    service = create_service(service_name='a - has free allowance')
-    template = create_template(service=service)
-    org = create_organisation(name="Org for {}".format(service.name))
+def test_fetch_usage_for_all_services_sms(notify_db_session):
+    service = set_up_yearly_data(service_name='Service 1')
```

Comment:
I'm not keen on big setup functions either. One positive outcome here is that we've moved away from the even-more-massive `set_up_usage_data` function, which is much harder to comprehend.

I think part of the pain you experienced here is the same as what I experienced trying to make any changes to these tests, especially the ones with bloated setup. I like that it's now easy to see what we're checking in each test due to the minimal setup, but I also agree that it hides some of subtlety of the data. I'm also worried these functions will go the same way as `set_up_usage_data` in future.

While we've agreed this is acceptable for now, I think it's worth making one final pass over this file:

- Consider restoring setup to some of the tests where it's only one or two lines.
- Switch to fixtures instead of calling a function, as this is more conventional.
- Break up the fixtures and make them really specific, so their use is clear in the test.

(These are added as stretch goals on the card.)